docs/headers.md

```diff
@@ -412,22 +412,21 @@ The `gsl::span` is based on the standardized version of `std::span` which was ad
 deprecate `gsl::span` when `std::span` finished standardization, however that plan changed when the runtime bounds checking
 was removed from `std::span`'s design.
 
-The only difference between `gsl::span` and `std::span` is that `gsl::span` strictly enforces runtime bounds checking.
-Any violations of the bounds check results in termination of the program.
-Like `gsl::span`, `gsl::span`'s iterators also differ from `std::span`'s iterator in that all access operations are bounds checked.
+The key differences between `gsl::span` and `std::span` are:
+- `gsl::span` strictly enforces runtime bounds checking for all access operations
+- Any violations of the bounds check results in termination of the program
+- `gsl::span`'s iterators also perform bounds checking, unlike `std::span`'s iterators
 
 #### Which version of span should I use?
 
-##### Use `gsl::span` if
+- C++14 & C++17 → Use `gsl::span` because it is your only option
+- C++20 & C++23 → Use `gsl::span` because accesses are bounds checked
+- C++26* → Use `gsl::span` because iterators are hardened too
 
-- you want to guarantee bounds safety in your project.
-  - All data accessing operations use bounds checking to ensure you are only accessing valid memory.
-- your project uses C++14 or C++17.
-  - `std::span` is not available as it was not introduced into the STL until C++20.
+\* C++26 or a backported standard library implementation.
 
-##### Use `std::span` if
-
-- your project is C++20 and you need the performance offered by `std::span`.
+Note that the behavior on bounds check violations is not configurable for GSL (`std::terminate()`), 
```

Comment:
Can we add a little more info here and make the style match the rest of the document?

Ideally there would be links to the hardening documentation for libc++, libstdc++, and STL. Also a link to the bounds-check violation handler in GSL may be useful.